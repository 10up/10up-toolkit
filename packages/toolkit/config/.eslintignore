build
dist
node_modules
vendor