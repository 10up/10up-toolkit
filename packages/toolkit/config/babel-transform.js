/**
 * External dependencies
 */
const babelJest = require('babel-jest');

module.exports = babelJest.default.createTransformer({
	presets: [[require.resolve('@10up/babel-preset-default'), { wordpress: true }]],
});
