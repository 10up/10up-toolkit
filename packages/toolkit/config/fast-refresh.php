<?php
// phpcs:ignoreFile
/**
 * This file is generated by 10up-toolkit and copied into your dist folder, do not modifies this file directly.
 */
namespace TenUpToolkit;

if ( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) {
	// This is development environment with a detected fast-refresh-file
	wp_die(
		sprintf(
			"You're using <a href='%s' target='_blank'>10up-toolkit</a>'s
			Hot Module Reloading but don't have <code>SCRIPT_DEBUG</code> enabled.<br/>
			Learn more about <a href='%s' target='_blank'>enabling HMR</a>.",
			"https://github.com/10up/10up-toolkit/tree/develop/packages/toolkit",
			"https://github.com/10up/10up-toolkit/tree/develop/packages/toolkit#hmr-and-fast-refresh"
		)
	);
}

if ( ! function_exists( __NAMESPACE__ . '\\set_dist_url_path' ) ) {
	$registry = [];

	function set_dist_url_path( $key, $url, $path ) {
		global $registry;

		$registry[ $key ] = [
			'dist_url' => $url,
			'dist_path' => $path,
		];

		global $dist_url, $dist_path;
	}

	function get_dist_url_path( $key ) {
		global $registry;

		return $registry[ $key ];
	}
}

add_action( 'init',
	/**
	 * Register HMR & React Fast Refresh scripts
	 *
	 * @return void
	 */
	function() {
		$project = basename( dirname( __DIR__ ) );
		$vars = get_dist_url_path( $project );

		$dist_path = $vars['dist_path'];
		$dist_url = $vars['dist_url'];

		if ( ! $dist_url || ! $dist_path ) {
			wp_die( '10up-toolkit error: you must defined call TenUpToolkit\set_dist_url_path with the URL and path to dist folfer to get fast refresh to work' );
		}

		$react_fast_refresh_entry   = $dist_url . 'fast-refresh/react-refresh-entry/index.min.js';
		$react_fast_refresh_runtime = $dist_url . 'fast-refresh/react-refresh-runtime/index.min.js';
		$hmr_runtime                = $dist_url . 'fast-refresh/hmr-runtime.js';

		$react_fast_refresh_entry_path   = $dist_path . 'fast-refresh/react-refresh-entry/index.min.js';
		$react_fast_refresh_runtime_path = $dist_path . 'fast-refresh/react-refresh-runtime/index.min.js';
		$hmr_runtime_path                = $dist_path . 'fast-refresh/hmr-runtime.js';

		if (
			! file_exists( $react_fast_refresh_entry_path )
			|| ! file_exists( $react_fast_refresh_runtime_path )
			|| ! file_exists( $hmr_runtime_path )
		) {
			return;
		}

		wp_register_script(
			'tenup-toolkit-react-fast-refresh-entry',
			$react_fast_refresh_entry,
			[],
			filemtime( $react_fast_refresh_entry_path ),
		);

		wp_register_script(
			'tenup-toolkit-react-refresh-runtime',
			$react_fast_refresh_runtime,
			[],
			filemtime( $react_fast_refresh_runtime_path ),
		);

		// the hmr runtime is unique to a 10up-toolkit build
		$prefix          = $project . '-';
		$hmr_runtime_key = $prefix . 'tenup-toolkit-hmr-runtime';

		wp_register_script(
			$hmr_runtime_key,
			$hmr_runtime,
			[],
			filemtime( $react_fast_refresh_runtime_path ),
		);

		global $wp_scripts;
		$script = $wp_scripts->query( 'react', 'registered' );

		if ( $script ) {
			// remove default react-refresh-entry by default
			if ( in_array( 'wp-react-refresh-entry', $script->deps ) ) {
				unset( $script->deps[ array_search( 'wp-react-refresh-entry', $script->deps ) ] );
			}

			if( !in_array( 'tenup-toolkit-react-refresh-runtime', $script->deps ) ){
				$script->deps[] = 'tenup-toolkit-react-refresh-runtime';
			}

			if( !in_array( 'tenup-toolkit-react-fast-refresh-entry', $script->deps ) ){
				$script->deps[] = 'tenup-toolkit-react-fast-refresh-entry';
			}

			if( !in_array( $hmr_runtime_key, $script->deps ) ){
				$script->deps[] = $hmr_runtime_key;
			}
		}
	},
	1
);


add_action(
	'wp_enqueue_scripts',
	function() {
		$prefix          = basename( dirname( __DIR__ ) ) . '-';
		$hmr_runtime_key = $prefix . 'tenup-toolkit-hmr-runtime';

		wp_enqueue_script( 'tenup-toolkit-react-refresh-runtime' );
		wp_enqueue_script( $hmr_runtime_key );
	},
	1
);

add_action(
	'admin_enqueue_scripts',
	function() {
		$prefix          = basename( dirname( __DIR__ ) ) . '-';
		$hmr_runtime_key = $prefix . 'tenup-toolkit-hmr-runtime';

		wp_enqueue_script( 'tenup-toolkit-react-refresh-runtime' );
		wp_enqueue_script( $hmr_runtime_key );
	},
	1
);


