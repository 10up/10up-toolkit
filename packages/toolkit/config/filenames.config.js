module.exports = {
	js: 'js/[name].js',
	jsChunk: 'js/[name].[contenthash].chunk.js',
	css: 'css/[name].css',
	block: 'blocks/[name]/editor.js',
	blockCSS: 'blocks/[name]/editor.css',
};
