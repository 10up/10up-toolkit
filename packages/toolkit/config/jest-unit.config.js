/**
 * External dependencies
 */
const path = require('path');

/**
 * Internal dependencies
 */
const { hasBabelConfig } = require('../utils');

const jestUnitConfig = {
	testMatch: ['**/__tests__/**/*.[jt]s?(x)', '**/test/*.[jt]s?(x)', '**/?(*.)test.[jt]s?(x)'],
	testPathIgnorePatterns: ['/node_modules/', '<rootDir>/vendor/'],
	timers: 'fake',
	setupFilesAfterEnv: ['@wordpress/jest-console'],
	moduleNameMapper: {
		'\\.(scss|css)$': path.join(__dirname, 'jest', 'style.mock.js'),
	},
	testEnvironment: 'jsdom',
};

if (!hasBabelConfig()) {
	jestUnitConfig.transform = {
		'^.+\\.[jt]sx?$': path.join(__dirname, 'babel-transform'),
	};
}

module.exports = jestUnitConfig;
