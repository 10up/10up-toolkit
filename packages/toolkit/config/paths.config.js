module.exports = {
	srcDir: './assets/',
	cssLoaderPaths: ['./assets/css', './includes/blocks'],
	copyAssetsDir: './assets/',
	blocksDir: './includes/blocks/',
};
