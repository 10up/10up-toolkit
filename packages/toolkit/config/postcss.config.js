const path = require('path');

module.exports = ({ file, env }) => {
	const config = {
		plugins: {
			'postcss-import': {},
			'postcss-mixins': {},
			'postcss-preset-env': {
				stage: 0,
				features: {
					'custom-properties': false,
				},
			},
		},
	};

	// Only load postcss-editor-styles plugin when we're processing the editor-style.css file.
	if (path.basename(file) === 'editor-style.css') {
		config.plugins['postcss-editor-styles'] = {
			scopeTo: '.editor-styles-wrapper',
			ignore: [':root', '.edit-post-visual-editor.editor-styles-wrapper', '.wp-toolbar'],
			remove: ['html', ':disabled', '[readonly]', '[disabled]'],
			tags: ['button', 'input', 'label', 'select', 'textarea', 'form'],
		};
	}

	config.plugins.cssnano =
		env === 'production'
			? {
					preset: [
						'default',
						{
							autoprefixer: false,
							calc: {
								precision: 8,
							},
							convertValues: true,
							discardComments: {
								removeAll: true,
							},
							mergeLonghand: false,
							zindex: false,
						},
					],
			  }
			: false;

	return config;
};
