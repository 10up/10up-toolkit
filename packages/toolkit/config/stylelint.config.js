const config = {
	extends: ['stylelint-config-standard-scss', '@10up/stylelint-config'],
};

module.exports = config;
