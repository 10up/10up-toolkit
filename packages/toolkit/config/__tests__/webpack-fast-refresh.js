import { getBuildFiles as getBuildFilesMock } from '../../utils/config';
import { hasProjectFile as hasProjectFileMock } from '../../utils/file';
import { getPackage as getPackageMock } from '../../utils/package';
import webpackSerializer from '../../test-utils/webpack-serializer';

jest.mock('../../utils/package', () => {
	const module = jest.requireActual('../../utils/package');

	jest.spyOn(module, 'getPackage');

	return module;
});

jest.mock('../../utils/config', () => {
	const module = jest.requireActual('../../utils/config');

	jest.spyOn(module, 'getBuildFiles');

	return module;
});

jest.mock('../../utils/file', () => {
	const module = jest.requireActual('../../utils/file');

	jest.spyOn(module, 'hasProjectFile');

	return module;
});

describe('webpack.config.js', () => {
	beforeAll(() => {
		expect.addSnapshotSerializer(webpackSerializer);
	});

	it('returns proper webpack fast refresh configs for project configs', () => {
		let webpackConfig;
		jest.isolateModules(() => {
			// eslint-disable-next-line global-require
			webpackConfig = require('../webpack-fast-refresh.config');
		});

		expect(webpackConfig).toMatchSnapshot();
	});

	it('includes react-webpack-fast-refresh with the --hot option', () => {
		process.argv.push('--hot');
		process.env.NODE_ENV = 'development';
		hasProjectFileMock.mockReturnValue(true);
		const entryBuildFiles = {
			entry1: 'entry1.js',
		};
		getBuildFilesMock.mockReturnValue(entryBuildFiles);
		getPackageMock.mockReturnValue({
			'10up-toolkit': {
				entry: entryBuildFiles,
			},
		});
		let webpackConfig;
		jest.isolateModules(() => {
			// eslint-disable-next-line global-require
			webpackConfig = require('../webpack.config');
		});

		expect(webpackConfig).toMatchSnapshot();
	});
});
