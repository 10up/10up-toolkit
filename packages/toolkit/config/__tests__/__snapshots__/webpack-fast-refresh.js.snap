// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack.config.js includes react-webpack-fast-refresh with the --hot option 1`] = `
Object {
  "devServer": Object {
    "allowedHosts": Array [
      ".test",
      ".local",
    ],
    "client": Object {
      "overlay": Object {
        "errors": true,
        "warnings": false,
      },
    },
    "devMiddleware": Object {
      "writeToDisk": true,
    },
    "hot": true,
    "port": 8000,
    "proxy": Object {
      "/dist": Object {
        "pathRewrite": Object {
          "^/dist": "",
        },
      },
    },
  },
  "devtool": "source-map",
  "entry": () => getEntryPoints(config),
  "experiments": Object {
    "outputModule": false,
  },
  "externals": Object {
    "jquery": "jQuery",
    "lodash": "lodash",
  },
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": input => shouldExclude(input, include),
        "test": /\\^\\(\\?!\\.\\*\\\\\\.d\\\\\\.tsx\\?\\$\\)\\.\\*\\\\\\.\\[tj\\]sx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/config/webpack/plugins/noop-loader.js",
          },
          Object {
            "loader": "/node_modules/babel-loader/lib/index.js",
            "options": Object {
              "cacheDirectory": true,
            },
          },
          Object {
            "loader": "@linaria/webpack-loader",
            "options": Object {
              "babelOptions": Object {},
              "extension": ".linaria.module.css",
              "overrideContext": context => ({
            ...context,
            $RefreshReg$: () => {}
          }),
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.svg\\$/,
        "use": Array [
          "@svgr/webpack",
          "url-loader",
        ],
      },
      Object {
        "exclude": Array [
          /\\\\\\.module\\\\\\.css\\$/,
          /\\\\\\.linaria\\\\\\.module\\\\\\.css/,
        ],
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          Object {
            "loader": "/node_modules/css-loader/dist/cjs.js",
            "options": Object {
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "/node_modules/postcss-loader/dist/cjs.js",
            "options": Object {
              "postcssOptions": Object {},
            },
          },
        ],
      },
      Object {
        "exclude": Array [
          /\\\\\\.module\\\\\\.css\\$/,
          /\\\\\\.linaria\\\\\\.module\\\\\\.css/,
        ],
        "test": /\\\\\\.\\(sc\\|sa\\)ss\\$/,
        "use": Array [
          Object {
            "loader": "/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          Object {
            "loader": "/node_modules/css-loader/dist/cjs.js",
            "options": Object {
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "/node_modules/postcss-loader/dist/cjs.js",
            "options": Object {
              "postcssOptions": Object {},
            },
          },
          Object {
            "loader": "/node_modules/sass-loader/dist/cjs.js",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "exclude": Array [
          /\\\\\\.linaria\\\\\\.module\\\\\\.css\\$/,
        ],
        "test": /\\\\\\.module\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          Object {
            "loader": "/node_modules/css-loader/dist/cjs.js",
            "options": Object {
              "import": false,
              "modules": true,
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "/node_modules/postcss-loader/dist/cjs.js",
            "options": Object {
              "postcssOptions": Object {},
            },
          },
          Object {
            "loader": "/node_modules/sass-loader/dist/cjs.js",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.linaria\\\\\\.module\\\\\\.css/,
        "use": Array [
          Object {
            "loader": "/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          Object {
            "loader": "css-loader",
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "concatenateModules": false,
    "minimizer": Array [
      TerserPlugin: {"test":{},"extractComments":true,"parallel":true,"minimizer":{"options":{"parse":{"ecma":8},"compress":{"ecma":5,"warnings":false,"comparisons":false,"inline":2}}}},
      ImageMinimizerPlugin: {"minimizer":{},"loader":true,"test":{},"deleteOriginalAssets":true},
      ImageMinimizerPlugin: {"minimizer":{},"loader":true,"test":{},"deleteOriginalAssets":true},
    ],
    "runtimeChunk": "single",
  },
  "output": Object {
    "chunkFilename": "js/[name].[contenthash].chunk.js",
    "clean": false,
    "filename": pathData => {
      if (pathData.chunk.name === 'runtime') {
        return 'fast-refresh/hmr-runtime.js';
      }
      // match windows and posix paths
      const isBlockAsset = buildFiles[pathData.chunk.name].match(/\\/blocks?\\//) || buildFiles[pathData.chunk.name].match(/\\\\blocks?\\\\/);
      return isBlockAsset ? filenames.block : filenames.js;
    },
    "path": "/dist",
    "publicPath": "/",
  },
  "performance": Object {
    "hints": "warning",
    "maxAssetSize": 10240000,
    "maxEntrypointSize": 40960000,
  },
  "plugins": Array [
    HtmlWebpackPlugin: {},
    ESLintWebpackPlugin: {"extensions":"js","emitError":true,"emitWarning":true,"failOnError":false,"resourceQueryExclude":[],"fix":false,"lintDirtyModulesOnly":true},
    MiniCssExtractPlugin: {"ignoreOrder":false,"runtime":true,"chunkFilename":"[id].css"},
    CopyPlugin: {},
    StylelintWebpackPlugin: {"cache":true,"cacheLocation":"node_modules/.cache/stylelint-webpack-plugin/.stylelintcache","extensions":["css","scss","sass"],"emitError":true,"emitWarning":true,"failOnError":false,"context":"/assets","files":"**/*.(s(c|a)ss|css)","allowEmptyInput":true,"lintDirtyModulesOnly":true},
    DependencyExtractionWebpackPlugin: {"combineAssets":false,"combinedOutputFile":null,"externalizedReport":false,"injectPolyfill":false,"outputFormat":"php","outputFilename":null,"useDefaults":true},
    CleanExtractedDeps: {},
    WebpackRemoveEmptyScriptsPlugin: {"enabled":true,"verbose":false,"extensions":{},"ignore":[],"remove":{}},
    TenUpToolkitTscPlugin: {},
    ReactRefreshPlugin: {"overlay":{"sockHost":"127.0.0.1","sockProtocol":"ws","sockPort":8000,"entry":"/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ErrorOverlayEntry.js","module":"/node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js","sockIntegration":"wds"},"exclude":[{},{}],"include":{}},
  ],
  "resolve": Object {
    "alias": Object {
      "lodash-es": "lodash",
    },
    "extensions": Array [
      ".tsx",
      ".ts",
      ".js",
    ],
  },
  "stats": Object {
    "all": false,
    "assets": true,
    "errorDetails": true,
    "errors": true,
    "excludeAssets": /\\\\\\.\\(jpe\\?g\\|png\\|gif\\|svg\\|woff\\|woff2\\)\\$/i,
    "moduleTrace": true,
    "modules": true,
    "performance": true,
    "warnings": true,
  },
  "target": "browserslist:> 1%, Firefox ESR, last 2 versions, not ie <= 11, not ie_mob <=11",
}
`;

exports[`webpack.config.js returns proper webpack fast refresh configs for project configs 1`] = `
Array [
  Object {
    "entry": Object {
      "react-refresh-entry": "/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js",
    },
    "mode": "development",
    "name": "react-refresh-entry",
    "output": Object {
      "filename": "[name]/index.min.js",
      "path": "/dist/fast-refresh",
    },
    "plugins": Array [
      DependencyExtractionWebpackPlugin: {"combineAssets":false,"combinedOutputFile":null,"externalizedReport":false,"injectPolyfill":false,"outputFormat":"php","outputFilename":null,"useDefaults":true},
    ],
  },
  Object {
    "entry": Object {
      "react-refresh-runtime": Object {
        "import": "react-refresh/runtime",
        "library": Object {
          "name": "ReactRefreshRuntime",
          "type": "window",
        },
      },
    },
    "mode": "development",
    "name": "react-refresh-runtime",
    "output": Object {
      "filename": "[name]/index.min.js",
      "path": "/dist/fast-refresh",
    },
    "plugins": Array [
      DependencyExtractionWebpackPlugin: {"combineAssets":false,"combinedOutputFile":null,"externalizedReport":false,"injectPolyfill":false,"outputFormat":"php","outputFilename":null,"useDefaults":false},
    ],
  },
]
`;
